curriculum/challenges/english/25-front-end-development/lab-user-configuration-manager/684aaf9ec670c68d20efd0d0.md

```diff
@@ -0,0 +1,384 @@
+---
+id: 684aaf9ec670c68d20efd0d0
+title: Build a User Configuration Manager
+challengeType: 27
+dashedName: build-a-user-configuration-manager
+---
+
+# --description--
+
+Fulfill the user stories below and get all the tests to pass to complete the lab.
+
+**User Stories:**
+
+1. You should create a dictionary named `settings` that would store the following user configuration preferences in this format:
+
+```md
+ "theme": "light",
+ "language": "English",
+ "notifications": "enabled"
+```
+
+1. You should define a function named `add_setting` that takes a dictionary of settings and a tuple containing a key-value pair as parameters.
+
+1. `add_setting(settings, (key, value))` should:
+
+   - Convert `key` to lowercase.
```

Comment:
This might be confuse some campers. so far the user stories tell them that the second parameter is a tuple representing a key-value pair. I am not sure how many campers will know to unpack these values from the tuple and then set the key to lowercase.

has there been tons of practice with unpacking values so far in the python curriculum? if not, some might struggle with what `key` is in reference to. 

so maybe more needs to be added here so they know `key` was unpacked from that tuple?

Reply:
Looking at the content prior to this lab, I see there are lecture videos for tuples. The workshop just before it also uses key, value and tuples. I think this should be good to go, but we can still ask @fhsinchy to comment on this.